examples/notificationReload.c

```diff
@@ -0,0 +1,190 @@
+/**
+ * @file
+ *
+ * @brief Example for notification library which reloads KDB when Elektra's
+ * configuration (e.g. mount points or global plugins) has changed.
+ *
+ * Requires:
+ *   - io_glib binding
+ *   - Transport plugins (e.g. kdb global-mount zeromqsend zeromqrecv && kdb run-hub-zeromq)
+ *
+ * Relevant keys for this example:
+ *   - /sw/elektra/example_notification/#0/current/value: Set to any integer value
+ *   Add additional transport plugins and remove the original pair afterwards or
+ *   mount a file which sets the key above to a different value and unmount it again
+ *
+ * @copyright BSD License (see LICENSE.md or https://www.libelektra.org)
+ *
+ */
+
+#include <kdb.h>
+#include <kdbhelper.h>       // elektraFree
+#include <kdbio.h>	   // I/O binding functions (elektraIo*)
+#include <kdbio/glib.h>      // I/O binding constructor for glib (elektraIoGlibNew)
+#include <kdbnotification.h> // notification functions
+
+#include <glib-unix.h> // g_unix_signal_add()
+#include <glib.h>      // glib functions
+
+#include <signal.h> // signal()
+#include <stdio.h>  // printf() & co
+#include <stdlib.h> // exit()
+
+#define TWO_SECONDS 2000
+#define RELOAD_INTERVAL 100
+
+GMainLoop * loop;
+
+KDB * kdb;
+Key * parentKey;
```

Comment:
are all of them needed globally? Otherwise the example is great!

Reply:
Thank you! No they can also be moved to a struct which is passed as user data - I updated the example to demo usage of the required functions (`elektraIo[Binding|Fd|Timer|Idle]GetData()`).

Reply:
Please check out the [updated example](https://github.com/ElektraInitiative/libelektra/pull/1850/files#diff-98f467ea2a20eaed5714de95cc0d4142).